addon/controllers/building/edit/dwellings.js

```diff
@@ -0,0 +1,42 @@
+import Controller from "@ember/controller";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { task } from "ember-concurrency-decorators";
+
+export default class BuildingEditDwellingsController extends Controller {
+  @service building;
+  @service dwelling;
+  @service intl;
+  @service notification;
+
+  @task
+  *fetchDwellings() {
+    try {
+      const building = yield this.building.getFromCacheOrApi(this.model);
+      return building.buildingEntrance.flatMap((entrance) =>
+        entrance.dwelling.map((dwelling) => {
+          dwelling.buildingEntranceNo = entrance.buildingEntranceNo;
+          return dwelling;
+        })
+      );
+    } catch (error) {
+      console.error(error);
+      this.notification.danger(
+        this.intl.t("ember-gwr.building.dwellings.error")
+      );
+    }
+  }
+
+  @action
+  async removeDwelling(dwelling) {
```

Comment:
Does it matter if deactivate is called multiple times?

Reply:
No i don't think so. Theoretically the list would be refreshed and the dwelling wouldn't even be there.

Reply:
But the @removeLink action is executed with every button click, couldn't it happen that it is executed multiple times? But I guess the worst that will happen that it throws an error on the subsequent tries.

Reply:
The problem with `dropTask` on this is, that if you would delete two models in a short time the drop task might drop the second delete even though its a different modell

Reply:
Oh I see, then how about a `disabled={{@removeLink.isRunning}}` on the `button`?

Reply:
Then all delete buttons are disabled and not just the button of the clicked model.

Reply:
Ok, then I guess not :sweat_smile: